src/lib/merkle_ledger_tests/test_database.ml

```diff
@@ -63,24 +67,36 @@ let%test_module "test functor on in memory databases" =
             | Some acct, Some acct' -> Account.equal acct acct'
             | _, _ -> false )
 
+      let%test_unit "set would change the merkle root" =
```

Comment:
I don't think we need this anymore. You just had this test to show me that set actually sets.

Reply:
I think I still need it, because I am about to change the implementation of `set_hash`, which was used in `set`.

Reply:
I still think we should remove this test because other tests fully test the behavior of `set`.